src/stake-pool/utils.ts

```diff
@@ -242,6 +242,90 @@ export async function calcWithdrawals(
   return res;
 }
 
+export async function calcWithdrawalsInverse(
```

Comment:
i just realized if we do the validator breakdown in this function as well we will need to update both `calcWithdrawals()` and this function if we change the validator selection algorithm. I think it might be ok to just export `calcWithdrawalReceiptInverse()` as `calcWithdrawalsInverse()` and then to actually use the result to generate the `ValidatorWithdrawalReceipt[]`, call `calcWithdrawals()` with the result's  `dropletsToUnstake`

Reply:
or we just call `calcWithdrawals()` here after calling `calcWithdrawalReceiptInverse()` lol

Reply:
I actually didn't understand much 😅 Not really familiar with how its being used, and what `ValidatorWithdrawalReceipts[]` are

Reply:
`calcWithdrawals()` takes `dropletsToUnstake` as an arg, calculates how many lamports is due and then withdraws those lamports across the stake pool's validators. Each `ValidatorWithdrawalReceipt` tells us how much lamports to withdraw as an active stake account from the given validator in the stake pool.

The current validator selection algorithm implemented in `calcWithdrawals()` simply withdraws the maximum possible stake from the stake pool's smallest validator and then moves to the next smallest validator if that's not enough.

For example, if our stake pool currently has 1 SOL staked to Validator A, 2 SOL staked to Validator B and 3 SOL staked to validator C, a withdrawal of 2 SOL will result in an array of 2 `ValidatorWithdrawalReceipt`s being returned: one to withdraw 1 SOL from Validator A and the other to withdraw 1 SOL from validator B. 

All this should be in the typedocs btw. I guess it isn't very clear on that front then? Do we need to revamp it?

So since `calcWithdrawalReceiptInverse()` gives you `dropletsUnstaked`, you can just `return calcWithdrawals(dropletsUnstaked, stakePoolAccount, validatorList);` at the end of the method